addons/sale/report/sale_report_views.xml

```diff
@@ -72,7 +72,7 @@
                     <filter string="Customer" name="Customer" context="{'group_by':'partner_id'}"/>
                     <filter string="Customer Country" name="country_id" context="{'group_by':'country_id'}"/>
                     <filter string="Customer Industry" name="industry_id" context="{'group_by':'industry_id'}"/>
-                    <filter string="Product" name="Product" context="{'group_by':'product_id'}"/>
+                    <filter string="Product" name="Product" context="{'group_by':'product_tmpl_id'}"/>
```

Comment:
Then we could probably add a "Product Variant" group by (only visible if variants are enabled) ;)

Reply:
I discussed it with @dalaOdoo and he said that since it's possible to do that in "Add custom group" then there is no need to add it in the main tab.

Reply:
@Feyensv yes but in master if variant is activated

Reply:
@dalaOdoo You mean, to leave only "Product" grouping in versions 16.0-16.4, and in master add "Product Variant" grouping if Variant option is selected?

Reply:
Let's do the improvement in master only (the customer will have to live with the current version, he would've to update his module to see the change anyway)